types/v1/types.go

```diff
@@ -81,6 +81,17 @@ type PersistentVolumeSource struct {
 //    Only one of its members may be specified. Currently OpenEBS is the only
 // member. There may be other members in future.
 type VolumeSpec struct {
+  // The context of this volume specification. 
+  // Examples: "controller", "replica". Implicitly inferred to be "replica" 
+  // if unspecified.
+  // +optional
+  Context VolumeContext `json:"context,omitempty" protobuf:"bytes,1,opt,name=context,casttype=VolumeContext"`
```

Comment:
Didn't get the context of `VolumeContext` here ?
How is going to help us, what is the purpose ?

Reply:
Yeah... This one came up while I was doing the policy stuff. This is tied to the the array of VolumeSpec. This will differentiate if the VolumeSpec is meant for Replica or Controller or something in future.

e.g. how are we going to know if we need to have 1 replica or 2 replicas for a controller ?